janus_core/calculations/md.py

```diff
@@ -442,6 +459,48 @@ def _restart_file(self) -> str:
             f"res-{step}.xyz", f"T{self.temp}", prefix_override=self.restart_stem
         )
 
+    def _parse_correlations(self):
+        """Parse correlation arguments into Correlations."""
+        if not self.correlation_kwargs:
+            self._correlations = None
+        elif len(self.correlation_kwargs["correlations"]) > 0:
+            if len(self.correlation_kwargs["correlation_parameters"]) != len(
+                self.correlation_kwargs["correlations"]
+            ):
+                raise ValueError("Mismatching sizes of correlations and parameters")
+
+            self._correlations = []
+            for cor in range(len(self.correlation_kwargs["correlations"])):
+                obs_a, obs_b = map(
+                    option_to_observable, self.correlation_kwargs["correlations"][cor]
+                )
+                params = self.correlation_kwargs["correlation_parameters"][cor]
+                self._correlations.append(Correlation(obs_a, obs_b, *params))
```

Comment:
Could even do:
```python
            for i, (obs_a, obs_b) in enumerate(map(option_to_observable, self.correlation_kwargs["correlations"])):
                params = self.correlation_kwargs["correlation_parameters"][i]
                self._correlations.append(Correlation(obs_a, obs_b, *params))
```
but you're very much getting into unreadable one-liner territory.
Could `zip` the `correlation_parameters` to avoid the `enumerate` though which makes it cleaner
```suggestion
            for params, (obs_a, obs_b) in zip(
                self.correlation_kwargs["correlation_parameters"]
                map(
                    option_to_observable, self.correlation_kwargs["correlations"])
                ):
                self._correlations.append(Correlation(obs_a, obs_b, *params))
```
At which point the thing could become a comprehension